crypto-core/src/main/java/com/palantir/crypto2/cipher/AesCtrCipher.java

```diff
@@ -80,30 +80,27 @@ public Cipher seek(long pos) {
         byte[] ivBytes = ivBuffer.toByteArray();
 
         // Ensure the iv is exactly BLOCK_SIZE bytes in length
-        final IvParameterSpec newIv;
         if (ivBytes.length >= IV_SIZE) {
-            newIv = new IvParameterSpec(ivBytes, ivBytes.length - IV_SIZE, IV_SIZE);
+            currIvParameterSpec = new IvParameterSpec(ivBytes, ivBytes.length - IV_SIZE, IV_SIZE);
         } else {
             final byte[] tmpIv = new byte[IV_SIZE];
             System.arraycopy(ivBytes, 0, tmpIv, IV_SIZE - ivBytes.length, ivBytes.length);
-            newIv = new IvParameterSpec(tmpIv);
+            currIvParameterSpec = new IvParameterSpec(tmpIv);
         }
 
-        Cipher cipher = getInstance();
-
         // Init the cipher with the new iv
         try {
-            cipher.init(currentOpmode, key, newIv);
-        } catch (InvalidKeyException | InvalidAlgorithmParameterException e) {
+            CryptoCipher cipher = CryptoCipherFactory.getCryptoCipher(ALGORITHM);
+            cipher.init(currentOpmode, key, currIvParameterSpec);
+
+            // Skip to the byte offset in the block where 'pos' is located
+            int bytesToSkip = (int) (pos % BLOCK_SIZE);
+            byte[] skip = new byte[bytesToSkip];
```

Comment:
should also leave a comment explaining the relationship between the sizes of skip/out and BLOCK_SIZE.